### PR TITLE
Move 'done' and 'req-done' events from sync to rs

### PR DESCRIPTION
**Pull request by galfert**
They are called 'sync-done' and 'sync-req-done' now.

refs #732

### CODE REVIEW

**Comment by gregkare** [comment]
It's missing the documentation changes in the List of events (doc/js-api/remotestorage.rst)

**Comment by galfert** [comment]
I added the documentation.

I noticed that the documentation also says that `remoteStorage` would emit the `wire-busy` and `wire-done` events as well. But they are emitted by `remoteStorage.remote` instead.

Should we move these events to `remoteStorage` as well?

**Comment by skddc** [comment]
@galfert Are they needed as public API? We're not using them in the widget, right?

File: doc/js-api/remotestorage.rst [hunk 1]
```diff
@@ -76,6 +76,14 @@ List of events
 """""""""""""
    Emitted when a wire request completes
 
+``sync-req-done``
+"""""""""""""""""
+   Emitted when a single sync task has finished
```
Comment (skddc) [inline doc/js-api/remotestorage.rst]: What does "sync task" mean exactly?

File: doc/js-api/remotestorage.rst [hunk 2]
```diff
@@ -76,6 +76,14 @@ List of events
...
+
+``sync-done``
+"""""""""""""
+   Emitted when the complete sync cycle has finished
```
Comment (skddc) [inline doc/js-api/remotestorage.rst]: What's "the" here? As this document describes the events emitted by rs in general, what is "the complete sync cycle"? I think this is supposed to mean a/one sync cycle/process?

**Comment by galfert** [comment]
Are they needed as public API? We're not using them in the widget, right?

They might be useful to use when caching is disabled for loading indication during up- and downloads.

**Comment by skddc** [comment]
Ah, right. So let's move them?

File: doc/js-api/remotestorage.rst (hunk 1 above)
Comment (galfert) [reply doc/js-api/remotestorage.rst]: If I understand correctly, a task can be a fetch of a directory or the upload of a local change.

Comment (skddc) [reply doc/js-api/remotestorage.rst]: So why not say "sync request" then? The event's name already uses it and seems to be correct and descriptive in this case.

**Comment by skddc** [comment]
(FYI: I linked the old issue about knowing when sync is done in the PR description)

**Comment by galfert** [comment]
Moved the wire events as well and wording of the documentation a bit.

File: doc/js-api/remotestorage.rst [hunk 3]
```diff
@@ -70,19 +70,19 @@ List of events
...
 
 ``wire-done``
 """""""""""""
-   Emitted when a wire request completes
+   Emitted when a network request completes
 
 ``sync-req-done``
 """""""""""""""""
-   Emitted when a single sync task has finished
+   Emitted when a sync request has finished
```
Comment (skddc) [inline doc/js-api/remotestorage.rst]: I actually liked the "single" in there. Makes it absolutely clear what is meant.

File: doc/js-api/remotestorage.rst [hunk 4]
```diff
@@ -70,19 +70,19 @@ List of events
...
 
 ``sync-done``
 """""""""""""
-   Emitted when the complete sync cycle has finished
+   Emitted when a sync cycle has finished
```
Comment (skddc) [inline doc/js-api/remotestorage.rst]: Again, not sure if removing "complete" is useful here, making it more ambiguous. I'd probably change "finished" to "completed" even, so it's clear when this is emitted. In a comment I found earlier, it also says "... and a new sync is scheduled", making it even clearer.

Comment (galfert) [reply doc/js-api/remotestorage.rst]: I like that description, will change it

**Comment by skddc** [comment]
LGTM

**Pull request merged**